src/js/views/maps/FilterByAttributeView.js

```diff
@@ -0,0 +1,285 @@
+        this.listenTo(
+          this.attributeSelect.model,
+          "change:selected",
+          (_model, attributeSelection) => {
+            this.handleAttributeChange(attributeSelection);
+          },
+        );
+      },
+
+      // Updates the attribute values select dropdown when any attribute/property is selected
+      handleAttributeChange(targetProperty) {
+        if (this.filters) {
+          const attributeValuesOptions = [];
+          this.filters.each((filterModel) => {
+            const property = filterModel.get("property");
+
+            if (targetProperty.includes(property)) {
+              const values = filterModel.get("allValues"); // Build the values dropdown with all "filterable" values for the selected property
+
+              if (Array.isArray(values)) {
+                values.forEach((val) => {
+                  attributeValuesOptions.push({
+                    label: val,
+                    value: val,
+                  });
+                });
+              }
+            }
+          });
+          // Update the attribute "values" dropdown with all values corresponding to the newly selected attribute
+          if (this.filterValuesSelection) {
+            this.filterValuesSelection.model.updateOptions(
+              attributeValuesOptions,
+            );
+            this.filterValuesSelection.model.set("selected", []);
+          }
+        }
+      },
+
+      /**
+       * Builds the atrribute values multi-select dropdown (using the selected filter attribute)
+       * Pulls filter attribute values from the map model.
+       * The attributes values are defined using the "allValues" key for each "property".
+       */
+      addAttributeValuesSelect() {
+        let defaultAttributeValues = []; // corresponds to values are defined using the "values" key for each "property" in the initial map model
+        const targetProperty = this.filterModel.get("property"); // TO DO: Get the selected attribute value from the attribute dropdown
+        if (this.model.get("visible")) {
+          defaultAttributeValues = this.preselectedAttributeValues;
+        }
+        const allValues = this.filterModel.get("allValues");
+        const attributeAllValues = [];
+        if (Array.isArray(allValues)) {
+          allValues.forEach((val) => {
+            attributeAllValues.push({
+              label: val,
+              value: val,
+            });
+          });
+        }
+
+        const valuesSelectContainer = this.$(
+          `.${this.classes.valuesDropdownContainer}`,
+        )[0];
+
+        if (!this.filterValuesSelection) {
+          const filterValueSelect = new SearchSelect({
+            options: attributeAllValues,
+            placeholderText: "Values",
+            inputLabel: "Select value(s)",
+            allowMulti: true,
+            allowAdditions: false,
+            separatorTextOptions: false,
+            selected: defaultAttributeValues,
```

Comment:
```suggestion
            selected: [...defaultAttributeValues],
```

wrapping an Array in a spread operator (`[... {array} ]`) makes a **duplicate** of the original array. We have to do this for the selected values that we pass to the SelectView because the SelectView modifies the array directly. Because arrays are [reference types](https://codeburst.io/explaining-value-vs-reference-in-javascript-647a975e12a0), when the SelectView modifies the array, it will modify the **original** array that's in the filter model. It does this silently, meaning there's no `"change:values"` event that gets triggered.

This can lead to unexpected behavior in the application.

When I was testing the SurfaceWater layer in the PDG map, the pre-selected value "0" wasn't showing up, *unil* I added the spread operator to make a copy of the array.